### DISCUSSION TITLE
Off-topic

### DISCUSSION THREAD

**Discussion by ghost** [RolandHughes/waylandgui#1] [General]
Can you enable Github Issues on your ls-cs repository?

I'm interested in your ls-cs project and want to learn more about it.

**Comment by RolandHughes** [comment]
I could, but it is not yet ready for others. I'm in the process of creating the official Web site, lscs-software.com. I live on a functioning family farm and harvest just started (rained this morning though) so that limits my time a bit.  I _had_ built generic Debian and RPM packages for CopperSpice. The Debian package was incorrect for multi-arch. To do embedded systems development using it one would have to build inside a Docker container. This *(*( won't allow an HTML attachment. I was going to send you the Manifesto. Ah, here. I may or may not make tweaks to [the Manifesto](https://lscs-software.com/LsCs-Manifesto.html). I may need to change the background on that, please tell me what you think.

I have the libraries going into x86_64-linux-gnu like they should. Waiting on a definitive answer from Debian about where /include files should go so that if you install -dev library for say 5 different architectures, then remove one, how do the include files survive?

The [Roadmap](https://lscs-software.com/LsCs-Roadmap.html) is nowhere near done, working on that now while doing system backups. I also seem to have pooched the stylesheet somehow because it has the correct color in BlueGriffon (what I'm using to create it.

I just pushed to ls-cs-1.9.1 branch what I have now. It will build a .Debian file in Debian 12. 

Warning: If you are using a limited computer, like an i7-gen4 it could take you 3 hours to build. Right now I'm building in a VM on an i9-gen13 with 128GB of RAM and "20" core. (Not certain I like e-cores).  I set the VM to 8-CPU and 24GB of RAM. I think it gets done in under 20 minutes.

You don't need a big expensive system though. One of my other dev machines is an ancient HP z820 with 20 physical core and 128GB of RAM. You can get those things [relatively cheap online](https://www.ebay.com/itm/184534473705?itmmeta=01J87T8SXG0CPZVRFWY21SZ6YF&hash=item2af71c9be9:g:KQsAAOSwCj5heayI&amdata=enc%3AAQAJAAAA4Mxmj%2BiGvOveHXEBClPb29jNmpogayfXEotkSjgM9e8SgTBa9CNpXUdDqZOnoSjKuHaXH%2BldR1gvPry4dmRUMHQZsLFk7cqlSAFDZfsOOZRxwgcIVvkWcrqm6egc5widyfopTOSs7vgIC6MAWJ0juxD%2BAg2gR0jQnGOTPe1wDCWz0b8NcHFW%2BNKBCkPjgWvmOv6HH8Sck0MQBLXqjW9jtp7Woa3a567k7FyUXnECyQFqDLNGSbA3iyLDlaYWsCfkn5c57lQrZ5DentOVYWXp082gO%2FSJF6dAOAq55RS4%2BTWc%7Ctkp%3ABFBM6p6j-sFk). Slap an SSD boot drive and a couple of fast spinning disks (for the VMs and builds) and for under $600 you have a kick bootie build machine. They are noisy, but I love them.

I do not recommend having it in the room where you sleep though as the fans aren't silent.

See, I'm old. I come from a time when we had to submit a compile to batch and wait for an operator to bring us the printout. While I can keep myself amused during a 3 hour build, I much prefer the under 30 minute build. <Grin>

Once you set up a Debian 12 VM in Oracle VirtualBox, clone the repo, then switch to the above mentioned branch.

`sudo ./LsCs-Deb-build-dependencies.sh`

after that works.

`sudo ./build-LsCs-deb.sh`

It will create a .deb but fail on the rename because I have too many plates spinning right now.

I haven't merged into mainline yet so . . . if you wanted to do anything you would have to branch my branch.

**If you were good/interested in RPM packaging**
You could ignore everything I said about Debian, take the RPM package building code I have, Get it to work with the new stuff, then research Multi-Arch packaging within RPM world and fix the package building. I have used both OpenSuse Leap and Fedora for RPM building using what is there prior to the LsCs fork.

![rip-out-qml-declarative](https://github.com/user-attachments/assets/e322cff4-6d72-4a82-9133-21b25ddd4aba)

The only other thing that could "safely" be done in an isolated branch right now is the attached image from the Roadmap.

Thank you for your interest and I welcome your help. Honestly, I didn't expect anyone but myself to care.

**Comment by RolandHughes** [comment]
I just finished a first hack at the Web site.

https://lscs-software.com/index.html

**Comment by ghost** [comment]
I don't think I can help you much. I'm only a high school student, and I don't have any expertise. I can help you test on MX Linux and Windows. MX Linux is what I have always used, and it's based on Debian.

Btw, please keep supporting old Windows. CopperSpice has declared that they moved to Windows 10 as the minimum supported version, even though their code can still compile and run on Windows 7.
